monitor/src/test/java/edu/asu/diging/monitor/core/service/impl/GroupManagerTest.java

```diff
@@ -0,0 +1,83 @@
+package edu.asu.diging.monitor.core.service.impl;
+
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.stubbing.OngoingStubbing;
+
+import edu.asu.diging.monitor.core.db.IGroupDbConnection;
+import edu.asu.diging.monitor.core.exceptions.GroupNotFoundException;
+import edu.asu.diging.monitor.core.exceptions.UnstorableObjectException;
+import edu.asu.diging.monitor.core.model.impl.App;
+import edu.asu.diging.monitor.core.model.impl.Group;
+
+public class GroupManagerTest {
+
+    @Mock
+    private IGroupDbConnection dbConnection;
+
+    @InjectMocks
+    private GroupManager managerToTest;
+
+    private List<Group> storedGroups;
+    private Group group1;
+    private Group group2;
+    private String ID1 = "GROUP1";
+    private String ID2 = "GROUP2";
+
+    @Before
+    public void setUp() throws GroupNotFoundException {
+        MockitoAnnotations.initMocks(this);
+        group1 = new Group();
+        group1.setId(ID1);
+        group2 = new Group();
+        group2.setId(ID2);
+        storedGroups = new ArrayList<Group>();
+        storedGroups.add(group1);
+        storedGroups.add(group2);
+
+        Mockito.when(dbConnection.getAllGroups()).thenReturn(storedGroups);
+        Mockito.when(dbConnection.getGroupById(ID1)).thenReturn(group1);
+        Mockito.when(dbConnection.getGroupById(ID2)).thenReturn(group2);
+    }
+
+    @Test
+    public void test_createGroup_success() throws UnstorableObjectException, GroupNotFoundException {
+        String id = "GROUP1";
+        Mockito.when(dbConnection.generateGroupId()).thenReturn(id);
+        Group group = managerToTest.createGroup(id);
+        Mockito.verify(dbConnection).createGroup(group);
+        Assert.assertEquals(id, group.getId());
+    }
+
+    @Test(expected = UnstorableObjectException.class)
+    public void test_createGroup_failure() throws UnstorableObjectException, GroupNotFoundException {
+        String id = "GROUP1";
+        Mockito.when(dbConnection.generateGroupId()).thenReturn(null);
+        Group group = managerToTest.createGroup(id);
+        Mockito.verify(dbConnection).createGroup(group);
```

Comment:
I owuldn't verify this here, since even if it doesn't call it as long as it throws an exception the test shoudl pass.